[DOCS][New Page] Creates topic for data views in Elastic Security (#2034)
elastic/security-docs: docs/getting-started/data-views-in-sec.asciidoc

[[data-views-in-sec]]

> = {Data-sources} in Elastic Security

Reviewer: I think the content under "The default data view" section would be good to include here it also talks about active data views.

Author: For now I tried adding an admonition to the top to direct readers to the info about the default. Let me know if you think we should do more!

The active {data-source} determines which data appears throughout {es-sec}.
{Data-sources} are defined by the index patterns they include.
Only data from the {es} {ref}/documents-indices.html[indices], {ref}/data-streams.html[data streams], or {ref}/alias.html[index aliases] specified in the active {data-source} will appear.

> == How to switch {data-sources}

Reviewer: ```suggestion
== Choose another {data-source}
```

You can tell which {data-source} is active by clicking the {Data-source} button that appears at the upper right of {es-sec} pages that display data, such as Overview, Alerts, or Hosts.

> To switch to another existing {data-source}, click the same button {Data-source} and choose a different {data-source} under `Choose {data-source}`:

Reviewer: ```suggestion
To switch to another {data-source}, go to `Choose {data-source}` and select one of the options.
```

> image::images/dataview-default-setting-highlighted.png[the securitySolution:defaultIndex setting, shown in context]

Reviewer: Think this image got accidentally switched around with the image in the "The default data view" section. :)

> == How to create or modify {data-sources}:

Reviewer: ```suggestion
== Create or modify {data-sources}
```

To modify or delete an existing {data-source} or to define a new one, you need the required permissions.
To learn more, refer to {apm-app-ref}/data-views.html[{kib} {Data-Sources}].

== The default {data-source}

The first time you add data to an {es-sec} instance within a given {apm-app-ref}/xpack-spaces.html[space], the default {data-source} gets generated for that space and becomes active.

> If you delete the active {data-source} when there are no other defined {data-sources}, the default {data-source} will regenerate and become active upon the refresh of any {es-sec} page in that space, because the app requires a {data-source} to function.

Reviewer: What do you think about putting this in an admonition to emphasize it more? I feel like this is something we want to call out, especially for users who are working with custom indices and data views.

Author: I think you're absolutely right that we should highlight this info for people working with custom indices. I added an admonition after the intro paragraph instead of in the "Default data view" section so it would be more prominent. What do you think?

The default {data-source} is defined by the `securitySolution:defaultIndex` setting, which you can modify from the Security Solution section of {kib}'s advanced settings (**Stack Management** > **Advanced Settings** > **Security Solution**):

image::images/dataview-button-highlighted.png[detail view of the Data view button]

The preset value for `securitySolution:defaultIndex` is (comma-delimited): `apm-*-transaction*,auditbeat-*,endgame-*,filebeat-*,logs-*,packetbeat-*,winlogbeat-*`.

> This includes, for example, all data shipped via {beats} or the {agent}.

Reviewer: You can probably do away with this information and link user to https://www.elastic.co/guide/en/security/master/advanced-settings.html#update-sec-indices instead. This section might also need to be refreshed a bit to mention how data views and index patterns intersect.

Also, if you do link to the "Advance settings" section, you can probably remove the `dataview-default-setting-highlighted.png` image since that topic already has a screenshot to show where the advanced Security settings are located.

Author: I think you're right, and that would be easier to maintain. I'm definitely open to it, but I'm torn about whether that would be too much linking. I feel like the page is a reasonable enough length that it might be okay to keep it in.

Reviewer 2: I'm in favor of linking to the existing content in Advanced Settings, for maintainability and to keep this new topic focused on the general idea of data views in the Security app.

[[data-views-in-sec]]
= {data-sources-cap} in Elastic Security

The active {data-source} determines what data appears throughout {es-sec} on pages with event or alert data.
{Data-sources} are defined by the index patterns they include.
Only data from the {es} {ref}/documents-indices.html[indices], {ref}/data-streams.html[data streams], or {ref}/alias.html[index aliases] specified in the active {data-source} will appear.

> IMPORTANT: Custom indices may not be included in the <<default-data-view-security, default {data source}>>. Modify it or create a custom {data-source} to include custom indices.

Reviewer 3: Looks like `<<default-data-view-security, default {data source}>>` is preventing variable substitution here?

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/171724727-c9165759-a404-4634-8f9a-915ec827c5fe.png" />
</p>

[discrete]
== Switch to another {data-source}

> You can tell which {data-source} is active by clicking the {data-source-cap} button that appears at the upper right of {es-sec} pages that display data, such as Overview, Alerts, or Hosts.

Reviewer 3: Should `Data View` always be capitalized here?

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/171725316-7e8436a3-4ef9-4d24-92b2-20f78dcbaa56.png" />
</p>

Author: Just when it's referring to the button I believe, since it should copy the capitalization of the button.

Reviewer 3: Ahh, yeah -- makes sense! Also seems the [main docs reference](https://www.elastic.co/guide/en/kibana/master/data-views.html) as lowercase.

> You can tell which {data-source} is active by clicking the {data-source-cap} button that appears at the upper right of {es-sec} pages that display data, such as Overview, Alerts, or Hosts.

Reviewer 3: > button that appears at the upper right of {es-sec} pages

Not sure if this is the spot to mention it, but Data View's are present within `Timeline`'s as well:

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/171726737-baabf100-78c5-4888-ad9f-ee61cac4e166.png" />
</p>

> You can tell which {data-source} is active by clicking the {data-source-cap} button that appears at the upper right of {es-sec} pages that display data, such as Overview, Alerts, Timelines, or Hosts.

Reviewer 2: A few tweaks here. I know the previous para also mentioned "event or alert data," but I think it's helpful to reiterate for clarity. To a new user, every page in the app probably looks like it contains some kind of data.
```suggestion
You can tell which {data-source} is active by clicking the *{data-source-cap}* menu at the upper right of {es-sec} pages that display event or alert data, such as Overview, Alerts, Timelines, or Hosts.
```

> To switch to another {data-source}, click `Choose {data-source}` and select one of the options.

Reviewer 3: Looks like `Choose {data-source}` is preventing variable substitution here?

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/171724969-63d66b44-23c4-43e9-9949-58261a6a7ae9.png" />
</p>

> To switch to another {data-source}, click `Choose {data-source}` and select one of the options.

Reviewer 4: might want to update the gif as well, the change will not take effect unless the Save button is pressed

image::images/dataview-select-gif.gif[gif of opening the data view selection menu]

> 

Reviewer 4: We should add a section about filtering the index patterns in a Data View. We should note that only index patterns that have matching indices with data are available for selection. The disabled (grayed out) index patterns do not currently match any indices.
![filtering](https://user-images.githubusercontent.com/6935300/171730149-9f949a74-110e-4b27-9627-b71190c9a568.gif)

> image::images/dataview-select-gif.gif[gif of opening the data view selection menu]

Reviewer 2: I think it's helpful to have an illustration here, but a static image would probably suffice -- all this is really showing is how to open a simple menu. A regular PNG with highlights could still illustrate the general location of the menu and what it looks like when opened, and it'd be easier to reproduce/crop/stitch in the future.

That said, the process that @stephmilovic mentions [below](https://github.com/elastic/security-docs/pull/2034/files#r888377300) is more complex, and so an animated GIF seems more appropriate to demonstrate that complexity.

[discrete]

> == Create or modify a {data-source}:

Reviewer 2: This section heading seems misleading -- it sounds like it's going to walk through a procedure, but the content is essentially just one sentence about permissions. Seems like this info should be in the introductory section, maybe as an admonition or prereqs.

```suggestion
== Create or modify a {data-source}
```

Author: I agree it doesn't walk through the steps, but it does link to them. I think it makes sense as its own section because users might find the page while trying to learn how to make a new data view, and the heading hopefully makes it easy to find while skimming. I don't think it would make sense as a prereq, because the page is an overview, not a tutorial, so there's no particular task described on the page to have prereqs. I do think it could work as admonition, but I'm not sure I see the downside to leaving it as-is, though I'm open to further feedback/suggestions.

Author: The section is a bit different after I incorporated some more feedback, so I'm curious what you think now.

Reviewer 2: Revised section is better! I still have a few more suggestions that I'll add separately from this thread, but the section now does what the heading implies.

To modify or delete an existing {data-source} or to define a new one, you need the required permissions.
To learn more, refer to {apm-app-ref}/data-views.html[{kib} {data-sources-cap}].

[discrete]
[[default-data-view-security]]
== The default {data-source}

> The first time you add data to an {es-sec} instance within a given {apm-app-ref}/xpack-spaces.html[space], the default {data-source} generates in that space and becomes active.

Reviewer 3: I believe the initial Data View isn't generated until the first page visit within the Security Solution app -- so if data is ingested but no user ever goes to the Security Solution app there won't be a Data View just yet. @stephmilovic can you verify here please?

> The first time a user visits Elastic Security within a given {apm-app-ref}/xpack-spaces.html[space], the default {data-source} generates in that space and becomes active.
> 
> If you delete the active {data-source} when there are no other defined {data-sources}, the default {data-source} will regenerate and become active upon the refresh of any {es-sec} page in the space, because the app requires a {data-source} to function.
> 
> The default {data-source} is defined by the `securitySolution:defaultIndex` setting, which you can modify from the Security Solution section of {kib}'s advanced settings (**Stack Management** > **Advanced Settings** > **Security Solution**):

Reviewer 2: I felt like this portion sort of buried the lead/lede, starting with a lot of internal workings before actually explaining what the default data source is. One idea is to invert the order a bit (below, along with a few other tweaks).

I also don't feel like we need to point out that the app requires a data source to function, because there isn't much actionable info from that. It'd be one thing if the user was required to re-create a data source to get it the app to work, but since the app does it automatically it's prob enough to just say that.

```suggestion
The default {data-source} is defined by the `securitySolution:defaultIndex` setting, which you can modify in {kib}'s advanced settings (**Stack Management** > **Advanced Settings** > **Security Solution**).

The first time a user visits {elastic-sec} within a given {apm-app-ref}/xpack-spaces.html[space] in Kibana, the default {data-source} generates in that space and becomes active.

If you delete the active {data-source} when there are no other defined {data-sources}, the default {data-source} will regenerate and become active upon the refresh of any {es-sec} page in the space.
```

Author: Love this reorg, thanks 👍 👍 👍

Only data from the {es} {ref}/documents-indices.html[indices], {ref}/data-streams.html[data streams], or {ref}/alias.html[index aliases] specified in the active {data-source} will appear.

IMPORTANT: Custom indices are not included in the <<default-data-view-security, default {data-source}>>. Modify it or create a custom {data-source} to include custom indices.

[discrete]
== Switch to another {data-source}

You can tell which {data-source} is active by clicking the *{data-source-cap}* menu at the upper right of {es-sec} pages that display event or alert data, such as Overview, Alerts, Timelines, or Hosts.
To switch to another {data-source}, click **Choose {data-source}**, select one of the options, and click **Save**:

image::images/dataview-button-highlighted.png[image highlighting how to open the data view selection menu]

[discrete]
== Create or filter a {data-source}

You can temporarily filter the active {data-source} from the *{data-source-cap}* menu by clicking *Advanced settings* and then adding or removing index patterns:

image::images/dataview-filter-example.gif[video showing how to filter the active data view]

> This method only allows you to add index patterns which match indices that currently contain data (other index patterns are unavailable and appear in grey). Note that changes made in this way are saved in the URL, and won't persist if you open a new tab.

Reviewer 5: ```suggestion
This only allows you to add index patterns that match indices that currently contain data (other index patterns are unavailable). Note that any changes made are saved in the current browser window and won't persist if you open a new tab.
```
